drivers/hmis/app/jobs/hmis/merge_clients_job.rb

```diff
@@ -109,29 +110,24 @@ def merge_custom_data_elements
 
         values = elements.sort_by(&:DateUpdated)
 
-        values[0..-2].each(&:destroy)
+        values[0..-2].each(&:destroy!)
       end
     end
 
-    def dedup(relation, keepers: [])
-      pairs = client_to_retain.send(relation).to_a.combination(2)
-
-      pairs.each do |pair|
-        equivalent = if pair[0].respond_to?(:equal_for_merge?)
-          pair[0].equal_for_merge?(pair[1])
+    def dedup(records, keepers: [])
```

Comment:
@ttoomey this looks correct to me. thank you for doing all this to fix the tests